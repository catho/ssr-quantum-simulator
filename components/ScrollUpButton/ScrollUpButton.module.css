.ScrollUpButton {
    position: fixed;
    bottom: 50px;
    right: 50px;
    height: 50px;
    z-Index: 9999999;
  }