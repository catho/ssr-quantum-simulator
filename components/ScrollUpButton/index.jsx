import { useEffect, useState } from 'react';
import { Button } from '@catho/quantum';
import styles from './ScrollUpButton.module.css';

const ScrollUpButton = () => {
  const [scrollUpButton, setScrollUpButton] = useState(false);
  useEffect(() => {
    window.addEventListener('scroll', () => {
      if (window.scrollY > 50) {
        setScrollUpButton(true);
      } else {
        setScrollUpButton(false);
      }
    });
  }, []);

  const scrollTop = () => {
    window.scrollTo({
      top: 0,
      behavior: 'auto',
    });
  };

  return (
    <>
      <div className={styles.ScrollUpButton}>
        {scrollUpButton && (
          <Button size="small" onClick={scrollTop}>
            Voltar ao Topo
          </Button>
        )}
      </div>
    </>
  );
};

export default ScrollUpButton;
