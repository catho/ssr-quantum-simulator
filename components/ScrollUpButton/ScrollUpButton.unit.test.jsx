import { render } from '@testing-library/react';
import ScrollUpButton from '.';

test('should match the snapshot', () => {
  const { container } = render(<ScrollUpButton />);
  expect(container).toMatchSnapshot();
});
