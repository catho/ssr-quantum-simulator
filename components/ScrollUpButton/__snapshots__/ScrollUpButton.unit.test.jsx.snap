// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div>
  <div
    class="ScrollUpButton"
  />
</div>
`;
